### PR TITLE
fix(web): boas-963 ensure start date is still displayed even when blank - Issue 1

### PR DESCRIPTION
**Pull request by richmoh**
## Issue ticket number and link
Issue 1 only of https://pins-ds.atlassian.net.mcas.ms/browse/BOAS-963

I spoke to @francescobrenci-AR and issue 2 is being fixed as part of another ticket
## Type of change 🧩

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

### CODE REVIEW

File: apps/web/src/server/applications/case/examination-timetable/applications-timetable.controller.js [hunk 1]
```diff
@@ -363,18 +363,25 @@ const getTimetableRows = (timetableItem) => {
...
 	return {
 		id,
 		itemTypeName: ExaminationTimetableType?.name,
 		name,
 		date: shouldShowField('date') ? displayDate(date, { condensed: true }) || '' : null,
-		startDate:
-			shouldShowField('startDate') && startDate
-				? displayDate(startDate, { condensed: true }) || ''
-				: null,
+		startDate: shouldShowField('startDate')
```
Comment (lutamKC) [inline apps/web/src/server/applications/case/examination-timetable/applications-timetable.controller.js:380]: a short-circuit evaluation (_condition ? option1 : option2_) within another short-circuit is usually very difficult to read: I suggest to rewrite this bit in a "longer" way, maybe above.

also, displayDate() doesn't accept an empty string as argument (you're gonna get the "Invalid time value" error), so this solution _displayDate(startDate || '')_ will not really work when startDate is not defined. 

maybe something like?
`displayDate(startDate, { condensed: true }) || ''`

File: apps/web/src/server/applications/case/examination-timetable/applications-timetable.controller.js [hunk 2]
```diff
@@ -363,18 +363,25 @@ const getTimetableRows = (timetableItem) => {
...
 	const templateType = ExaminationTimetableType?.templateType;
 
+	if (!templateType) {
+		throw new Error(
+			`Template type not found for timetable item type ${ExaminationTimetableType?.name}`
+		);
+	}
+
 	const shouldShowField = (/** @type {string} */ fieldName) =>
-		Object.prototype.hasOwnProperty.call(timetableTemplatesSchema[templateType || 0], fieldName);
```
[the remark this answers is not in the document]
Comment (lutamKC) [reply apps/web/src/server/applications/case/examination-timetable/applications-timetable.controller.js:367]: templateType should be always be defined as it's necessary field, but I guess an extra check won't hurt :)

**Review by lutamKC — CHANGES_REQUESTED** [review]

File: apps/web/src/server/applications/case/examination-timetable/applications-timetable.controller.js (hunk 1 above)
Comment (richmoh) [reply apps/web/src/server/applications/case/examination-timetable/applications-timetable.controller.js:380]: Totally. Will make the relevant changes.

File: apps/web/src/server/applications/case/examination-timetable/applications-timetable.controller.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (lutamKC) [reply apps/web/src/server/applications/case/examination-timetable/applications-timetable.controller.js]: I made some changes to the interface of TimetableItem. 
Now the field "templateType" is recognized as mandatory and always defined and that ugly "`templateType || 0`" is not necessary anymore.

**Review by lutamKC — APPROVED** [review]

**Pull request merged**